crates/server/src/store/metrics.rs

```diff
@@ -31,7 +37,55 @@ impl PersistentStoreMetrics {
             "number of dns zone records in persisted storages"
         );
 
-        Self { zone_records }
+        #[cfg(feature = "__dnssec")]
+        let (
+            zone_records_dynamically_added,
+            zone_records_dynamically_deleted,
+            zone_records_dynamically_updated,
+        ) = {
+            let zone_records_dynamically_modified_name =
+                "hickory_zone_records_dynamically_modified_total";
+
+            let operation_key = "operation";
+
+            let records_added = counter!(zone_records_dynamically_modified_name, store_key => store, operation_key => "added");
+            let records_deleted = counter!(zone_records_dynamically_modified_name, store_key => store, operation_key => "deleted");
+            let records_updated = counter!(zone_records_dynamically_modified_name, store_key => store, operation_key => "updated");
+
+            describe_counter!(
+                zone_records_dynamically_modified_name,
+                Unit::Count,
+                "number of dns zone records that had been dynamically modified"
+            );
+
+            (records_added, records_deleted, records_updated)
+        };
+
+        Self {
+            zone_records,
+            #[cfg(feature = "__dnssec")]
+            zone_records_dynamically_added,
+            #[cfg(feature = "__dnssec")]
+            zone_records_dynamically_deleted,
+            #[cfg(feature = "__dnssec")]
+            zone_records_dynamically_updated,
+        }
+    }
+
+    #[cfg(feature = "__dnssec")]
+    pub(super) fn dynamically_add(&self) {
```

Comment:
Prefer `added()`/`deleted()`/`updated()` rather than the active `add()` (etc) for these.

Reply:
Updated.